### PR TITLE
Make compilation options for akka typed modules strict

### PR DESCRIPTION
**Pull request by chbatey**
i'll add a bunch more of the other typed modules into this PR

### CODE REVIEW

**Comment by raboof** [comment]
refs #26088, on top of #26552

**Review by raboof — APPROVED** [review]

File: akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala [hunk 1]
```diff
@@ -48,7 +48,7 @@ class DnsDiscoverySpec extends AkkaSpec(DnsDiscoverySpec.config) with DockerBind
 
   val systemWithAsyncDnsAsResolver = ActorSystem("AsyncDnsSystem", configWithAsyncDnsResolverAsDefault)
 
-  private def testSrvRecords(discovery: ServiceDiscovery): Unit = {
+  private def testSrvRecords(discovery: ServiceDiscovery) = {
```
Comment (patriknw) [inline akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala]: or end the method with `()`? (only a test so doesn't matter)

File: akka-stream-typed/src/test/scala/akka/stream/typed/scaladsl/CustomGuardianAndMaterializerSpec.scala [hunk 2]
```diff
@@ -6,29 +6,14 @@ package akka.stream.typed.scaladsl
...
-  case object Complete extends GuardianProtocol
-  case object Failed extends GuardianProtocol
-}
-
 class CustomGuardianAndMaterializerSpec extends ScalaTestWithActorTestKit with WordSpecLike {
-  import CustomGuardianAndMaterializerSpec._
-
-  val guardian = Behaviors.receive[GuardianProtocol] { (_, msg) =>
-    Behaviors.same
-  }
```
Comment (patriknw) [inline akka-stream-typed/src/test/scala/akka/stream/typed/scaladsl/CustomGuardianAndMaterializerSpec.scala]: I wonder what the original purpose of this test was? Perhaps rewrite `"work with typed ActorSystem with custom guardian"` to actually start another ActorSystem with this guardian?

File: akka-stream-typed/src/test/scala/docs/akka/stream/typed/ActorSourceSinkExample.scala [hunk 3]
```diff
@@ -44,7 +45,10 @@ object ActorSourceSinkExample {
     // #actor-source-ref
   }
 
+
   {
+    def targetActor(): ActorRef[Protocol] = ???
```
Comment (patriknw) [inline akka-stream-typed/src/test/scala/docs/akka/stream/typed/ActorSourceSinkExample.scala]: not a warning here?

File: project/AkkaDisciplinePlugin.scala [hunk 4]
```diff
@@ -66,14 +70,14 @@ object AkkaDisciplinePlugin extends AutoPlugin with ScalafixSupport {
   /**
     * Remain visibly filtered for future code quality work and removing.
     */
-  val undisciplineScalacOptions = Seq(
+  val undisciplineScalacOptions = Set(
```
Comment (patriknw) [inline project/AkkaDisciplinePlugin.scala]: why change? isn't it anyway converted to a Seq?

File: akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala [hunk 5]
```diff
@@ -126,15 +126,17 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
...
+        val x = ctx.spawnAnonymous[Nothing](Behaviors.setup[Nothing] { innerCtx =>
           innerCtx.watch(target)
-          Behaviors.receive[Nothing]((_, _) => Behaviors.same).receiveSignal {
+          Behaviors.receiveSignal[Nothing] {
             case (_, Terminated(`target`)) =>
               ctx.self ! msg
               Behaviors.stopped
           }
         })
+        println(x)
```
Comment (patriknw) [inline akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala]: leftover

File: akka-cluster-typed/src/test/scala/akka/cluster/ddata/typed/scaladsl/ReplicatorSpec.scala [hunk 6]
```diff
@@ -103,12 +101,12 @@ object ReplicatorSpec {
       val replicator: ActorRef[Replicator.Command] = ???
       implicit val timeout = Timeout(3.seconds)
       implicit val scheduler: Scheduler = ???
-      implicit val cluster: Cluster = ???
+      implicit val cluster: SelfUniqueAddress= ???
```
Comment (patriknw) [inline akka-cluster-typed/src/test/scala/akka/cluster/ddata/typed/scaladsl/ReplicatorSpec.scala]: fmt?

File: akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala (hunk 1 above)
Comment (chbatey) [reply akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala]: I've removed this option for tests now as it is very common

File: akka-stream-typed/src/test/scala/akka/stream/typed/scaladsl/CustomGuardianAndMaterializerSpec.scala (hunk 2 above)
Comment (chbatey) [reply akka-stream-typed/src/test/scala/akka/stream/typed/scaladsl/CustomGuardianAndMaterializerSpec.scala]: Yes i was curious, and the actor system does have a custom guardian as it is the test kit's one

File: akka-stream-typed/src/test/scala/docs/akka/stream/typed/ActorSourceSinkExample.scala (hunk 3 above)
Comment (chbatey) [reply akka-stream-typed/src/test/scala/docs/akka/stream/typed/ActorSourceSinkExample.scala]: It was a dead code warning for the code below the ???, i've now removed dead code warnings from ytest

File: project/AkkaDisciplinePlugin.scala (hunk 4 above)
Comment (chbatey) [reply project/AkkaDisciplinePlugin.scala]: to use `--`, could do filter

File: akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala (hunk 5 above)
Comment (chbatey) [reply akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala]: Doh, was playing around trying to get this warning free

File: project/AkkaDisciplinePlugin.scala (hunk 4 above)
Comment (patriknw) [reply project/AkkaDisciplinePlugin.scala]: I have no opinion

File: akka-stream-typed/src/test/scala/docs/akka/stream/typed/ActorSourceSinkExample.scala [hunk 7]
```diff
@@ -55,16 +58,20 @@ object ActorSourceSinkExample {
...
       ActorSink.actorRef[Protocol](ref = actor, onCompleteMessage = Complete, onFailureMessage = Fail.apply)
 
     Source.single(Message("msg1")).runWith(sink)
     // #actor-sink-ref
+
   }
 
   {
+
+    def targetActor(): ActorRef[Protocol] = ???
```
Comment (helena) [inline akka-stream-typed/src/test/scala/docs/akka/stream/typed/ActorSourceSinkExample.scala]: is this really a side effect needing `def targetActor(): ActorRef[Protocol]` versus def targetActor: ActorRef[Protocol]?

File: project/AkkaDisciplinePlugin.scala [hunk 8]
```diff
@@ -21,7 +21,7 @@ object AkkaDisciplinePlugin extends AutoPlugin with ScalafixSupport {
   override def requires: Plugins = JvmPlugin && ScalafixPlugin
   override lazy val projectSettings = disciplineSettings
   
-  val strictProjects = Set("akka-discovery")
+  val strictProjects = Set("akka-discovery", "akka-stream-typed", "akka-cluster-typed")
```
Comment (helena) [inline project/AkkaDisciplinePlugin.scala]: The idea originally was everything was strict and this qualification would not be needed, so eventually, that's the goal.

Comment (helena) [reply project/AkkaDisciplinePlugin.scala]: But that work is a module by module process :)

**Review by helena — APPROVED** [review]
LGTM

**Comment by chbatey** [comment]
Leaving this until after fatal is removed for docs generation

**Pull request merged**